tdvt/tdvt/config_gen/tdvtconfig.py

```diff
@@ -4,33 +4,43 @@
 import re
 
 from ..resources import *
+from .test_config import TestConfig, RunTimeTestConfig
 
-class TdvtTestConfig(object):
+class TdvtInvocation(object):
     """Track how items were tested. This captures how tdvt was invoked."""
-    def __init__(self, tested_sql=False, tested_tuples=True, tds='', config='', output_dir='', logical=False, verbose=False, override='', suite_name='', from_args=None, thread_count=6, from_json=None, run_as_perf=False, timeout_seconds=60 * 60):
-        self.tested_sql = tested_sql
-        self.tested_tuples = tested_tuples
+    def __init__(self, from_args=None, from_json=None, test_config : TestConfig = None):
+        self.tested_sql = False
+        self.tested_tuples = True
         self.log_dir = ''
-        self.output_dir = output_dir
-        self.timeout_seconds = timeout_seconds
-        self.logical = logical
-        self.config_file = config
-        self.suite_name = suite_name
-        self.d_override = override
-        self.verbose = verbose
+        self.output_dir = ''
+        self.timeout_seconds = 60 * 60
+        self.logical = False
+        self.config_file = ''
+        self.suite_name = ''
+        self.d_override = ''
+        self.verbose = False
         self.command_line = ''
         self.noheader = False
-        self.thread_count = thread_count
+        self.thread_count = 6
         self.leave_temp_dir = False
-        self.run_as_perf = run_as_perf
+        self.run_as_perf = False
         self.thread_id = -1
         self.tds = ''
+        self.tested_run_time_config = None
+
         if from_args:
             self.init_from_args(from_args)
         if from_json:
             self.init_from_json(from_json)
-        if tds:
-            self.tds = tds
+        if test_config:
+            self.set_run_time_test_config(test_config.run_time_config)
+            self.suite_name = test_config.dsname
+
+    def set_run_time_test_config(self, rtt):
```

Comment:
would this be the correct type hint to include here?

```suggestion
    def set_run_time_test_config(self, rtt: RunTimeTestConfig):
```